tests/python/unittest/test_text.py

```diff
@@ -119,49 +119,33 @@ def test_indices_to_tokens():
 
     assertRaises(ValueError, indexer.to_tokens, 100)
 
+def test_download_embed():
+    @TokenEmbedding.register
+    class Test(TokenEmbedding):
+        pretrained_file_name_sha1 = \
+            {'embedding_test.vec': '29b9a6511cf4b5aae293c44a9ec1365b74f2a2f8'} # 33 bytes
+        namespace = 'test'
 
-def test_glove():
-    glove_6b_50d = TokenEmbedding.create(
-        'glove', pretrained_file_name='glove.6B.50d.txt')
+        def __init__(self, embedding_root=os.path.join('~', '.mxnet', 'embeddings'),
+                     init_unknown_vec=nd.zeros, **kwargs):
+            pretrained_file_name = 'embedding_test.vec'
+            Test._check_pretrained_file_names(pretrained_file_name)
 
-    assert len(glove_6b_50d) == 400001
-    assert glove_6b_50d.vec_len == 50
-    assert glove_6b_50d.token_to_idx['hi'] == 11084
-    assert glove_6b_50d.idx_to_token[11084] == 'hi'
+            super(Test, self).__init__(**kwargs)
 
-    first_vec_sum = glove_6b_50d.idx_to_vec[0].sum().asnumpy()[0]
-    assert_almost_equal(first_vec_sum, 0)
+            pretrained_file_path = Test._get_pretrained_file(embedding_root,
+                                                             pretrained_file_name)
 
-    unk_vec_sum = glove_6b_50d.get_vecs_by_tokens(
-        '<unk$unk@unk>').sum().asnumpy()[0]
-    assert_almost_equal(unk_vec_sum, 0)
+            self._load_embedding(pretrained_file_path, ' ', init_unknown_vec)
 
-    unk_vecs_sum = glove_6b_50d.get_vecs_by_tokens(
-        ['<unk$unk@unk>', '<unk$unk@unk>']).sum().asnumpy()[0]
-    assert_almost_equal(unk_vecs_sum, 0)
+    test_embed = TokenEmbedding.create('test')
```

Comment:
To be consistent with common use cases, such as
glove_6b_50d = TokenEmbedding.create('glove', pretrained_file_name='glove.6B.50d.txt')
fasttext_simple = TokenEmbedding.create('fasttext', pretrained_file_name='wiki.simple.vec', init_unknown_vec=nd.ones)

Can we provide both arguments embedding_name ('test') + pretrained_file_name ('embedding_test.vec').

Reply:
This test is intended for testing the download and integrity verification logic. Whether it has pretrained file name as argument or hard-coded shouldn't affect the result.